src/Driver/SolverTypes/HEVISolverType.jl

```diff
@@ -0,0 +1,199 @@
+
+    linear_max_subspace_size::Int
+    linear_atol::FT
+    linear_rtol::FT
+
+    nonlinear_max_iterations::Int
+    nonlinear_rtol::FT
+    nonlinear_ϵ::FT
+
+    preconditioner_update_freq::Int
+
+    function HEVISolverType(
+        FT;
+        solver_method = ARK2ImplicitExplicitMidpoint,
+        linear_max_subspace_size = Int(30),
+        linear_atol = FT(-1.0),
+        linear_rtol = FT(5e-5),
+        nonlinear_max_iterations = Int(10),
+        nonlinear_rtol = FT(1e-4),
+        nonlinear_ϵ = FT(1.e-10),
+        preconditioner_update_freq = Int(50),
+    )
+
+        return new{FT}(
+            solver_method,
+            linear_max_subspace_size,
+            linear_atol,
+            linear_rtol,
+            nonlinear_max_iterations,
+            nonlinear_rtol,
+            nonlinear_ϵ,
+            preconditioner_update_freq,
+        )
+    end
+end
+
+"""
+    getdtmodel(ode_solver::HEVISolverType, bl)
+
+A function which returns a model representing the dynamics
+with the most restrictive time-stepping requirements.
+"""
+function getdtmodel(ode_solver::HEVISolverType, bl)
+    # Most restrictive dynamics are treated implicitly
+    return bl
+end
+
+"""
+# Description
+    solversetup(
+        ode_solver::HEVISolverType{FT},
+        dg,
+        Q,
+        dt,
+        t0,
+        diffusion_direction,
+    )
+
+Creates an ODE solver using a HEVI-type time-integration
+scheme. All horizontal acoustic waves are treated explicitly,
+while the 1-D vertical problem is treated implicitly. All
+other dynamics (advection, diffusion) is treated explicitly
+in the additive Runge-Kutta method.
+
+# Comments:
+Currently, the only HEVI-type splitting ClimateMachine can currently
+do only involves splitting the acoustic processes; it is not currently
+possible to perform more fine-grained separation of tendencies
+(for example, including vertical advection or diffusion in the 1-D implicit problem)
+"""
+function solversetup(
+    ode_solver::HEVISolverType,
+    dg,
+    Q,
+    dt,
+    t0,
+    diffusion_direction,
+)
+
+
+    # All we need to do is create a DGModel for the
+    # vertical acoustic waves (determined from the `implicit_model`)
+    vdg = DGModel(
+        dg.balance_law,
+        dg.grid,
+        dg.numerical_flux_first_order,
+        dg.numerical_flux_second_order,
+        dg.numerical_flux_gradient;
+        state_auxiliary = dg.state_auxiliary,
+        direction = VerticalDirection(),
+        diffusion_direction = VerticalDirection(),
+    )
+
+    # linear solver relative tolerance rtol which should be slightly smaller than the nonlinear solver tol
+    linearsolver = BatchedGeneralizedMinimalResidual(
+        dg,
+        Q;
+        max_subspace_size = ode_solver.linear_max_subspace_size,
+        atol = ode_solver.linear_atol,
+        rtol = ode_solver.linear_rtol,
+    )
+
+    """
+    N(q)(Q) = Qhat  => F(Q) = N(q)(Q) - Qhat
+
+    F(Q) == 0
+    ||F(Q^i) || / ||F(Q^0) || < tol
+
+    """
```

Comment:
Use multiline comments instead of strings
```suggestion
    #=
    N(q)(Q) = Qhat  => F(Q) = N(q)(Q) - Qhat

    F(Q) == 0
    ||F(Q^i) || / ||F(Q^0) || < tol
    =#
```